src/tests/max_element.hpp

```diff
@@ -0,0 +1,119 @@
+// ========================== MAX ELEMENT TESTS ============================= //
+// Project:         The Experimental Bit Algorithms Library
+// Name:            max_element.hpp
+// Description:     Tests for max_element function overloads for bits 
+// Creator:         Vincent Reverdy
+// Contributor(s):  Collin Gress [2019]
+// License: BSD 3-Clause License
+// ========================================================================== //
+#ifndef _MAX_ELEMENT_TESTS_HPP_INCLUDED
+#define _MAX_ELEMENT_TESTS_HPP_INCLUDED
+// ========================================================================== //
+
+
+
+// =============================== PREAMBLE ================================= //
+// C++ standard library
+#include <iostream>
+// Project sources
+#include "test_root.cc"
+#include "bit.hpp"
+// Third-party libraries
+// Miscellaneous
+// ========================================================================== //
+
+
+
+// --------------------------- Max Element Tests ---------------------------- //
```

Comment:
I think that since we have the framework and tools in place to, we should be testing each algorithm using the template product so that we can test on each of the applicable iterator types as well.